pyls/workspace.py

```diff
@@ -218,6 +232,7 @@ def word_at_position(self, position):
 
         return m_start[0] + m_end[-1]
 
+    @lock
     def jedi_names(self, all_scopes=False, definitions=True, references=False):
```

Comment:
Why does `jedi_names` need to be locked, but not the other Jedi operations?

Reply:
Since you added the `lock` decorator to `jedi_script`, does the one above this line is still necessary?

Reply:
What do you think about what I said in my comment above? Do you really need to add the `lock` decorator to `jedi_names`?